aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,190 @@
+```
+
+Note: The preceding output my differ from what you see depending on the version of  .NET 8 used.
+Review the contents of output directory:
+
+```
+PS> dir bin\Release\net8.0\win-x64\publish
+
+    Directory: C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\publish
+
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a---          30/03/2023  1:41 PM        9480704 MyFirstAotWebApi.exe
+-a---          30/03/2023  1:41 PM       43044864 MyFirstAotWebApi.pdb
+```
+
+The API template with AOT enabled produces a binary of about 9.4MB on Windows, although the size may vary depending on the build of .NET 8.0 used. The executable can be run to a machine without the .NET Core runtime installed.
+
+```
+PS> .\bin\Release\net8.0\win-x64\publish\MyFirstAotWebApi.exe
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://localhost:5000
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Production
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: C:\Code\Demos\MyFirstAotWebApi
+```
+
+The `Program.cs` source file contains some changes for publishing to .NET native AOT:
+
+```csharp
+using System.Text.Json.Serialization;
+using MyFirstAotWebApi;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+builder.Logging.AddConsole();
+
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+});
+
+var app = builder.Build();
+
+var sampleTodos = TodoGenerator.GenerateTodos().ToArray();
+
+var todosApi = app.MapGroup("/todos");
+todosApi.MapGet("/", () => sampleTodos);
+todosApi.MapGet("/{id}", (int id) =>
+    sampleTodos.FirstOrDefault(a => a.Id == id) is { } todo
+        ? Results.Ok(todo)
+        : Results.NotFound());
+
+app.Run();
+
+[JsonSerializable(typeof(Todo[]))]
+internal partial class AppJsonSerializerContext : JsonSerializerContext
+{
+
+}
+```
+
+A significant difference is that `Microsoft.AspNetCore.Builder.WebApplication.CreateBuilderSlim` is used to create the web application builder.  The `CreateBuilderSlim` method initializes the <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> with incompatible ASP.NET Core features that are disabled.
+<!-- Update the preceding with the following when the .NET 8 API is published :
+<xref:Microsoft.AspNetCore.Builder.WebApplication.CreateBuilderSlim%2A>
+-->
+
+Native AOT is unable to use reflection at runtime. Source generators are used to produce code to avoid the need for reflection. In some cases source generators produce code optimized for AOT even when a generator is not strictly required. To view source code that is generated based on the code in ```Program.cs``` modify the ```MyFirstAotWebApi.csproj``` to include the ```<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>``` property. Example:
```

Comment:
```suggestion
Because unused code is trimmed during publishing for Native AOT, the application cannot use unbounded reflection at runtime. Source generators are used to produce code to avoid the need for reflection. In some cases source generators produce code optimized for AOT even when a generator is not strictly required. To view source code that is generated based on the code in ```Program.cs``` modify the ```MyFirstAotWebApi.csproj``` to include the ```<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>``` property. Example:
```